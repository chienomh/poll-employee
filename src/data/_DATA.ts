import {
  IQuestions,
  ISaveQuestionReq,
} from "../components/list-questions/list-question.model";
import { ISaveQuestionAnsweredReq } from "../pages/detail-question-page/detail-question.model";
import { IUsers } from "../pages/login-page/login.model";

let users: IUsers = {
  sarahedo: {
    id: "sarahedo",
    password: "password123",
    name: "Sarah Edo",
    avatarURL:
      "https://static.vecteezy.com/system/resources/thumbnails/002/002/427/small/man-avatar-character-isolated-icon-free-vector.jpg",
    answers: {
      "8xf0y6ziyjabvozdd253nd": "optionOne",
      "6ni6ok3ym7mf1p33lnez": "optionOne",
      am8ehyc8byjqgar0jgpub9: "optionTwo",
      loxhs1bqm25b708cmbf3g: "optionTwo",
    },
    questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
  },
  tylermcginnis: {
    id: "tylermcginnis",
    password: "abc321",
    name: "Tyler McGinnis",
    avatarURL:
      "https://static.vecteezy.com/system/resources/thumbnails/002/002/403/small/man-with-beard-avatar-character-isolated-icon-free-vector.jpg",
    answers: {
      vthrdm985a262al8qx3do: "optionOne",
      xj352vofupe1dqz9emx13r: "optionTwo",
    },
    questions: ["loxhs1bqm25b708cmbf3g", "vthrdm985a262al8qx3do"],
  },
  mtsamis: {
    id: "mtsamis",
    password: "xyz123",
    name: "Mike Tsamis",
    avatarURL:
      "https://as2.ftcdn.net/v2/jpg/02/23/50/73/1000_F_223507324_jKl7xbsaEdUjGr42WzQeSazKRighVDU4.jpg",
    answers: {
      xj352vofupe1dqz9emx13r: "optionOne",
      vthrdm985a262al8qx3do: "optionTwo",
      "6ni6ok3ym7mf1p33lnez": "optionOne",
    },
    questions: ["6ni6ok3ym7mf1p33lnez", "xj352vofupe1dqz9emx13r"],
  },
  zoshikanlu: {
    id: "zoshikanlu",
    password: "pass246",
    name: "Zenobia Oshikanlu",
    avatarURL:
      "https://www.shutterstock.com/image-vector/young-man-avatar-character-260nw-661669825.jpg",
    answers: {
      xj352vofupe1dqz9emx13r: "optionOne",
    },
    questions: [],
  },
};

let questions: any = {
  "8xf0y6ziyjabvozdd253nd": {
    id: "8xf0y6ziyjabvozdd253nd",
    author: "sarahedo",
    timestamp: 1467166872634,
    optionOne: {
      votes: ["sarahedo"],
      text: "Build our new application with Javascript",
    },
    optionTwo: {
      votes: [],
      text: "Build our new application with Typescript",
    },
  },
  "6ni6ok3ym7mf1p33lnez": {
    id: "6ni6ok3ym7mf1p33lnez",
    author: "mtsamis",
    timestamp: 1468479767190,
    optionOne: {
      votes: [],
      text: "hire more frontend developers",
    },
    optionTwo: {
      votes: ["mtsamis", "sarahedo"],
      text: "hire more backend developers",
    },
  },
  am8ehyc8byjqgar0jgpub9: {
    id: "am8ehyc8byjqgar0jgpub9",
    author: "sarahedo",
    timestamp: 1488579767190,
    optionOne: {
      votes: [],
      text: "conduct a release retrospective 1 week after a release",
    },
    optionTwo: {
      votes: ["sarahedo"],
      text: "conduct release retrospectives quarterly",
    },
  },
  loxhs1bqm25b708cmbf3g: {
    id: "loxhs1bqm25b708cmbf3g",
    author: "tylermcginnis",
    timestamp: 1482579767190,
    optionOne: {
      votes: [],
      text: "have code reviews conducted by peers",
    },
    optionTwo: {
      votes: ["sarahedo"],
      text: "have code reviews conducted by managers",
    },
  },
  vthrdm985a262al8qx3do: {
    id: "vthrdm985a262al8qx3do",
    author: "tylermcginnis",
    timestamp: 1489579767190,
    optionOne: {
      votes: ["tylermcginnis"],
      text: "take a course on ReactJS",
    },
    optionTwo: {
      votes: ["mtsamis"],
      text: "take a course on unit testing with Jest",
    },
  },
  xj352vofupe1dqz9emx13r: {
    id: "xj352vofupe1dqz9emx13r",
    author: "mtsamis",
    timestamp: 1493579767190,
    optionOne: {
      votes: ["mtsamis", "zoshikanlu"],
      text: "deploy to production once every two weeks",
    },
    optionTwo: {
      votes: ["tylermcginnis"],
      text: "deploy to production once every month",
    },
  },
};

function generateUID() {
  return (
    Math.random().toString(36).substring(2, 15) +
    Math.random().toString(36).substring(2, 15)
  );
}

export function _getUsers() {
  const data = new Promise<IUsers>((resolve) => {
    setTimeout(() => resolve({ ...users }), 1000);
  });
  return data;
}

export function _getQuestions() {
  return new Promise<IQuestions>((resolve) => {
    setTimeout(() => resolve({ ...questions }), 1000);
  });
}

function formatQuestion({ optionOneText, optionTwoText, author }: any) {
  return {
    id: generateUID(),
    timestamp: Date.now(),
    author,
    optionOne: {
      votes: [],
      text: optionOneText,
    },
    optionTwo: {
      votes: [],
      text: optionTwoText,
    },
  };
}

export function _saveQuestion(question: ISaveQuestionReq) {
  return new Promise((resolve, reject) => {
    if (
      !question.optionOneText ||
      !question.optionTwoText ||
      !question.author
    ) {
      reject("Please provide optionOneText, optionTwoText, and author");
    }

    const formattedQuestion = formatQuestion(question);
    setTimeout(() => {
      questions = {
        ...questions,
        [formattedQuestion.id]: formattedQuestion,
      };

      resolve(formattedQuestion);
    }, 1000);
  });
}

export function _saveQuestionAnswer({
  authedUser,
  qid,
  answer,
}: ISaveQuestionAnsweredReq) {
  return new Promise((resolve, reject) => {
    if (!authedUser || !qid || !answer) {
      reject("Please provide authedUser, qid, and answer");
    }

    setTimeout(() => {
      if (authedUser) {
        users = {
          ...users,
          [authedUser]: {
            ...users[authedUser],
            answers: {
              ...users[authedUser].answers,
              [qid]: answer,
            },
          },
        };

        questions = {
          ...questions,
          [qid]: {
            ...questions[qid],
            [answer]: {
              ...questions[qid][answer],
              votes: questions[qid][answer].votes.concat([authedUser]),
            },
          },
        };
      }
      resolve(true);
    }, 500);
  });
}
