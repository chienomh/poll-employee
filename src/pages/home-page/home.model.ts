import { IQuestions } from "../../components/list-questions/list-question.model";

export interface IHomeState {
  listQuestion?: IQuestions;
  isLoading: string;
}
