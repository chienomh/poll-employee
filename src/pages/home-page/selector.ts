import { RootState } from "../../store";

export const getAllQuestion = (state: RootState) => state.home.listQuestion;
