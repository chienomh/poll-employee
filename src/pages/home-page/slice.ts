import { PayloadAction, createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { IHomeState } from "./home.model";
import { IQuestions } from "../../components/list-questions/list-question.model";
import { _getQuestions } from "../../data/_DATA";

const initialState: IHomeState = {
  isLoading: "",
  listQuestion: undefined,
};

export const fetchAllQuesttion = createAsyncThunk(
  "fetch-all-question",
  async () => {
    const res: IQuestions = await _getQuestions();
    return res;
  }
);

const homeSlicer = createSlice({
  name: "home-slice",
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(
      fetchAllQuesttion.fulfilled,
      (state: IHomeState, action: PayloadAction<IQuestions>) => {
        state.listQuestion = action.payload;
      }
    );
  },
});

// export const { getAllQuestion } = homeSlicer.actions;
export default homeSlicer.reducer;
