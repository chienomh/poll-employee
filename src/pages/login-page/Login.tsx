import { useEffect, useState } from "react";
import { Button, Container, Form, Image } from "react-bootstrap";
import { useNavigate } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../../store";
import { IFormData } from "./login.model";
import "./login.scss";
import { getListUsers } from "./selector";
import { fetchAllUsers, setCurrentUser } from "./slice";

export default function Login() {
  const [formData, setFormData] = useState<IFormData>({
    password: "",
    user: "",
  });

  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const listUser = useAppSelector(getListUsers);

  const handleChangeInput = (e: React.FocusEvent<HTMLInputElement>) => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value,
    });
  };

  const handleLogin = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    if (listUser) {
      for (const key in listUser) {
        if (
          key === formData.user &&
          listUser[key].password === formData.password
        ) {
          dispatch(setCurrentUser(listUser[key]));
          break;
        }
      }
    }
    navigate("/");
  };

  useEffect(() => {
    dispatch(fetchAllUsers());
  }, []);
  return (
    <Container>
      <div className="wrapper-login">
        <div className="header">Login Employee Polls</div>
        <Image
          src="https://bc.hocthionline.net/wp-content/uploads/2023/04/Login.jpg"
          alt="this is login image"
          width={500}
        />
        <div className="form">
          <Form>
            <Form.Label htmlFor="inputUser5">User</Form.Label>
            <Form.Control
              type="text"
              id="inputUser5"
              aria-describedby="passwordHelpBlock"
              onBlur={handleChangeInput}
              name="user"
              data-testid="user"
            />
            <Form.Label htmlFor="inputPassword5">Password</Form.Label>
            <Form.Control
              type="Password"
              id="inputPassword5"
              aria-describedby="passwordHelpBlock"
              onBlur={handleChangeInput}
              name="password"
              data-testid="password"
            />
            <div
              style={{
                display: "flex",
                justifyContent: "center",
                marginTop: "20px",
              }}
            >
              <Button
                variant="primary"
                style={{ width: "200px" }}
                type="submit"
                onClick={handleLogin}
              >
                Login
              </Button>
            </div>
          </Form>
        </div>
      </div>
    </Container>
  );
}
