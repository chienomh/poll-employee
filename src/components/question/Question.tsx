import { Button } from "react-bootstrap";
import { IQuestion } from "../list-questions/list-question.model";
import "./question.scss";
import { useNavigate } from "react-router-dom";
import { useAppSelector } from "../../store";
import { getCurrentUser } from "../../pages/login-page/selector";

export default function Question(props: IQuestion) {
  const navigate = useNavigate();
  const currentUser = useAppSelector(getCurrentUser);

  const options: Intl.DateTimeFormatOptions = {
    hour: "2-digit",
    minute: "2-digit",
    hour12: true,
    month: "2-digit",
    day: "2-digit",
    year: "numeric",
  };

  const date = new Intl.DateTimeFormat("en-US", options).format(
    props.timestamp
  );

  const handleClickShow = () => {
    if (currentUser) {
      navigate(`/questions/${props.id}`);
    } else {
      navigate("/login");
    }
  };
  return (
    <div className="wrapper-question">
      <div className="info-question">
        <div className="title-question">{props.author}</div>
        <div className="time-question">{date}</div>
      </div>
      <div className="show-button">
        <Button
          variant="outline-success"
          style={{ width: "100%" }}
          onClick={handleClickShow}
        >
          Show
        </Button>
      </div>
    </div>
  );
}
