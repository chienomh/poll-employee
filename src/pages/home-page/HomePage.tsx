import { useEffect, useMemo } from "react";
import ListQuestion from "../../components/list-questions/ListQuestion";
import { IQuestions } from "../../components/list-questions/list-question.model";
import { useAppDispatch, useAppSelector } from "../../store";
import { getCurrentUser } from "../login-page/selector";
import { fetchAllUsers } from "../login-page/slice";
import "./home.scss";
import { getAllQuestion } from "./selector";
import { fetchAllQuesttion } from "./slice";

export const HomePage = () => {
  const dispatch = useAppDispatch();
  const listQuestion = useAppSelector(getAllQuestion);
  const currentUser = useAppSelector(getCurrentUser);

  const filterQuestion = useMemo(() => {
    let dataAnswered: IQuestions = {};
    let dataUnanswered: IQuestions = {};

    if (currentUser) {
      for (const key in listQuestion) {
        if (
          listQuestion[key].optionOne.votes.includes(currentUser?.id) ||
          listQuestion[key].optionTwo.votes.includes(currentUser?.id)
        ) {
          dataAnswered[key] = listQuestion[key];
        } else {
          dataUnanswered[key] = listQuestion[key];
        }
      }
    } else {
      dataUnanswered = { ...listQuestion };
    }
    return [dataUnanswered, dataAnswered];
  }, [listQuestion]);

  useEffect(() => {
    dispatch(fetchAllQuesttion());
    dispatch(fetchAllUsers());
  }, []);
  return (
    <div className="wrapper-content">
      <ListQuestion title="New Question" questions={filterQuestion[0]} />
      <ListQuestion title="Done" questions={filterQuestion[1]} />
    </div>
  );
};
