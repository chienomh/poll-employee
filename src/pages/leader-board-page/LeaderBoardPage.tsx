import { Image, Table } from "react-bootstrap";
import { useAppSelector } from "../../store";
import { getCurrentUser, getListUsers } from "../login-page/selector";
import { IUser } from "../login-page/login.model";
import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

const handleSort = (a: IUser, b: IUser) => {
  const totalA = a.questions.length + Object.keys(a.answers).length;
  const totalB = b.questions.length + Object.keys(b.answers).length;
  return totalA > totalB ? -1 : 1;
};

export default function LeaderBoardPage() {
  const [users, setUsers] = useState<IUser[]>([]);
  const listUsers = useAppSelector(getListUsers);
  const currentUser = useAppSelector(getCurrentUser);
  const navigate = useNavigate();
  useEffect(() => {
    const filterUser: IUser[] = [];
    if (listUsers && currentUser) {
      Object.values(listUsers).forEach((item) => filterUser.push(item));
      setUsers(filterUser.sort((a, b) => handleSort(a, b)));
    } else {
      navigate("/login");
    }
  }, [listUsers]);

  return (
    <div>
      <Table striped bordered hover>
        <thead>
          <tr>
            <th>Users</th>
            <th>Answered</th>
            <th>Created</th>
          </tr>
        </thead>
        <tbody>
          {users.map((item) => (
            <tr key={item.id}>
              <td style={{ display: "flex", gap: "10px" }}>
                {item.avatarURL && (
                  <Image src={item.avatarURL} roundedCircle width={50} />
                )}
                <div>
                  <div style={{ fontWeight: "bold" }}>{item.name}</div>
                  <div>{item.id}</div>
                </div>
              </td>
              <td>{Object.keys(item.answers).length}</td>
              <td>{item.questions.length}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
}
