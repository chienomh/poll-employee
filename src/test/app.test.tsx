import "@testing-library/jest-dom";
import { fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Login from "src/pages/login-page/Login";
import { renderWithProviders } from "./test-utils";
import { HomePage } from "src/pages/home-page/HomePage";
import Question from "src/components/question/Question";
import LeaderBoardPage from "src/pages/leader-board-page/LeaderBoardPage";
import { _saveQuestion, _saveQuestionAnswer } from "src/data/_DATA";

describe("Test my application", () => {
  test("render application", () => {
    const { getByText } = renderWithProviders(
      <Router>
        <Login />
      </Router>
    );

    const title = getByText("Login Employee Polls");
    expect(title).toBeInTheDocument();
  });

  test("Test login success", () => {
    const { getByTestId, getByDisplayValue, getByRole } = renderWithProviders(
      <Router>
        <Login />
      </Router>
    );
    const userInput = getByTestId("user");
    const passwordInput = getByTestId("password");

    expect(userInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();

    fireEvent.change(userInput, { target: { value: "sarahedo" } });
    fireEvent.change(passwordInput, { target: { value: "password123" } });

    const userInputValue = getByDisplayValue("sarahedo");
    expect(userInputValue).toBeInTheDocument();

    const loginBtn = getByRole("button", { name: "Login" });
    expect(loginBtn).toBeInTheDocument();

    fireEvent.click(loginBtn);
    expect(window.location.pathname).toBe("/");
  });

  test("Test home", () => {
    const { getByText } = renderWithProviders(
      <Router>
        <HomePage />
      </Router>,
      {
        preloadedState: {
          home: {
            listQuestion: {
              am8ehyc8byjqgar0jgpub9: {
                id: "am8ehyc8byjqgar0jgpub9",
                author: "sarahedo",
                timestamp: 1488579767190,
                optionOne: {
                  votes: [],
                  text: "conduct a release retrospective 1 week after a release",
                },
                optionTwo: {
                  votes: ["sarahedo"],
                  text: "conduct release retrospectives quarterly",
                },
              },
            },
            isLoading: "",
          },
        },
      }
    );

    const titleListQuestion = getByText("New Question");
    expect(titleListQuestion).toBeInTheDocument();

    const authorQuestion = getByText("sarahedo");
    expect(authorQuestion).toBeInTheDocument();
  });

  test("Should navigate /login if if the user is not logged in", () => {
    const { getByRole } = renderWithProviders(
      <Router>
        <Question
          author="sarahedo"
          id="123123123"
          optionOne={{
            votes: [],
            text: "conduct a release retrospective 1 week after a release",
          }}
          optionTwo={{
            votes: ["sarahedo"],
            text: "conduct release retrospectives quarterly",
          }}
          timestamp={1482579767190}
        />
      </Router>
    );

    const showBtn = getByRole("button", { name: "Show" });
    fireEvent.click(showBtn);

    expect(window.location.pathname).toBe("/login");
  });

  test("Should navigate /question/:question_id if if the user is logged in", () => {
    const { getByRole } = renderWithProviders(
      <Router>
        <Question
          author="sarahedo"
          id="123123123"
          optionOne={{
            votes: [],
            text: "conduct a release retrospective 1 week after a release",
          }}
          optionTwo={{
            votes: ["sarahedo"],
            text: "conduct release retrospectives quarterly",
          }}
          timestamp={1482579767190}
        />
      </Router>,
      {
        preloadedState: {
          login: {
            isLoading: "",
            users: undefined,
            currentUser: {
              id: "sarahedo",
              password: "password123",
              name: "Sarah Edo",
              avatarURL:
                "https://static.vecteezy.com/system/resources/thumbnails/002/002/427/small/man-avatar-character-isolated-icon-free-vector.jpg",
              answers: {
                "8xf0y6ziyjabvozdd253nd": "optionOne",
                "6ni6ok3ym7mf1p33lnez": "optionOne",
                am8ehyc8byjqgar0jgpub9: "optionTwo",
                loxhs1bqm25b708cmbf3g: "optionTwo",
              },
              questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
            },
          },
        },
      }
    );

    const showBtn = getByRole("button", { name: "Show" });
    fireEvent.click(showBtn);

    expect(window.location.pathname).toBe("/questions/123123123");
  });

  test("Should show leader board when user logged in", () => {
    const { getByText } = renderWithProviders(
      <Router>
        <LeaderBoardPage />
      </Router>,
      {
        preloadedState: {
          login: {
            isLoading: "",
            users: {
              sarahedo: {
                id: "sarahedo",
                password: "password123",
                name: "Sarah Edo",
                avatarURL:
                  "https://static.vecteezy.com/system/resources/thumbnails/002/002/427/small/man-avatar-character-isolated-icon-free-vector.jpg",
                answers: {
                  "8xf0y6ziyjabvozdd253nd": "optionOne",
                  "6ni6ok3ym7mf1p33lnez": "optionOne",
                  am8ehyc8byjqgar0jgpub9: "optionTwo",
                  loxhs1bqm25b708cmbf3g: "optionTwo",
                },
                questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
              },
            },
            currentUser: {
              id: "sarahedo",
              password: "password123",
              name: "Sarah Edo",
              avatarURL:
                "https://static.vecteezy.com/system/resources/thumbnails/002/002/427/small/man-avatar-character-isolated-icon-free-vector.jpg",
              answers: {
                "8xf0y6ziyjabvozdd253nd": "optionOne",
                "6ni6ok3ym7mf1p33lnez": "optionOne",
                am8ehyc8byjqgar0jgpub9: "optionTwo",
                loxhs1bqm25b708cmbf3g: "optionTwo",
              },
              questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
            },
          },
        },
      }
    );

    const title = getByText("Answered");
    expect(title).toBeInTheDocument();
  });

  test("Should show navigate /login board when user not logged in", () => {
    renderWithProviders(
      <Router>
        <LeaderBoardPage />
      </Router>
    );

    expect(window.location.pathname).toBe("/login");
  });

  test("Save question success", async () => {
    const response = await _saveQuestion({
      optionOneText: "React",
      optionTwoText: "Vuejs",
      author: "sarahedo",
    });

    expect(response).toBeTruthy();
  });

  test("save question faild", async () => {
    const response = await _saveQuestion({
      optionOneText: "React",
      optionTwoText: "Vuejs",
      author: undefined,
    }).catch((error) => error);

    expect(response).toBe(
      "Please provide optionOneText, optionTwoText, and author"
    );
  });

  test("save question answer success", async () => {
    const response = await _saveQuestionAnswer({
      answer: "optionOne",
      qid: "xj352vofupe1dqz9emx13r",
      authedUser: "sarahedo",
    });
    expect(response).toBeTruthy();
  });

  test("save question answer fail", async () => {
    const response = await _saveQuestionAnswer({
      answer: "optionOne",
      qid: "123123",
      authedUser: 0,
    }).catch((error) => error);
    expect(response).toBe("Please provide authedUser, qid, and answer");
  });

  test("render snapshot", () => {
    const component = renderWithProviders(
      <Router>
        <LeaderBoardPage />
      </Router>
    );

    expect(component).toMatchSnapshot();
  });
});
