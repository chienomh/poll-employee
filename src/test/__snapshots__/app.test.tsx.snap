// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my application render snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Users
              </th>
              <th>
                Answered
              </th>
              <th>
                Created
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Users
            </th>
            <th>
              Answered
            </th>
            <th>
              Created
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
